test/multiple-user-principal-types.test.js

```diff
@@ -54,338 +69,295 @@ describe('Multiple users with custom principalType:', function() {
+      return OneUser.login(commonCredentials)
+        .then(function(accessToken) {
+          assertGoodToken(accessToken, userFromOneModel);
+        });
     });
 
-    it('Login a user from 2nd user model by providing credentials', function(done) {
-      AnotherUser.login(commonCredentials)
-      .then(function(accessToken) {
-        assertGoodToken(accessToken, userFromAnotherModel);
-        done();
-      })
-      .catch(done);
+    it('Login a user from 2nd user model by providing credentials', function() {
+      return AnotherUser.login(commonCredentials)
+        .then(function(accessToken) {
+          assertGoodToken(accessToken, userFromAnotherModel);
+        });
     });
 
-    it('Login should only allow correct credentials', function(done) {
-      OneUser.login({email: 'foo@bar.com', password: 'invalid'})
-      .then(function(accessToken) {
-        expect(accessToken, 'accessToken').to.not.exist();
-        done();
-      })
-      .catch(function(err) {
-        expect(err, 'err').to.exist();
-        expect(err).to.have.property('code', 'LOGIN_FAILED');
-        done();
-      });
+    it('Login should only allow correct credentials', function() {
+      return OneUser.login({email: 'foo@bar.com', password: 'invalid'})
+        .then(
+          function onSuccess() {
+            throw new Error('OneUser.login() should have failed');
+          },
+          function onError(err) {
+            expect(err).to.have.property('code', 'LOGIN_FAILED');
+          }
+        );
     });
   });
 
   function assertGoodToken(accessToken, user) {
     if (accessToken instanceof AccessToken) {
       accessToken = accessToken.toJSON();
     }
+    expect(accessToken.id, 'token id').to.have.lengthOf(64);
     expect(accessToken).to.have.property('userId', user.id);
     expect(accessToken).to.have.property('principalType', user.constructor.definition.name);
-    assert(accessToken.id);
-    assert.equal(accessToken.id.length, 64);
   }
 
   describe('User.logout', function() {
     it('should logout a user from user model 1 without logging out user from model 2',
-    function(done) {
-      var userFromOneModelTokens, userFromAnotherModelTokens;
-      Promise.all([
+    function() {
+      var tokenOfOneUser, tokenOfAnotherUser;
+      return Promise.all([
         OneUser.login(commonCredentials),
         AnotherUser.login(commonCredentials),
       ])
-      .spread(function(at1, at2) {
-        userFromOneModelTokens = at1;
-        userFromAnotherModelTokens = at2;
-        return OneUser.logout(userFromOneModelTokens.id);
+      .spread(function(t1, t2) {
+        tokenOfOneUser = t1;
+        tokenOfAnotherUser = t2;
+        return OneUser.logout(tokenOfOneUser.id);
       })
       .then(function() {
         return Promise.all([
-          AccessToken.findById(userFromOneModelTokens.id),
-          AccessToken.findById(userFromAnotherModelTokens.id),
+          AccessToken.findById(tokenOfOneUser.id),
```

Comment:
Perhaps we should search for all tokens and then assert that an expected array was returned?

IMO, assertions like `exists` and `not.exists` are difficult to troubleshoot when they fail, because the error message provides no context.